zkevm-circuits/src/table.rs

```diff
@@ -942,19 +944,19 @@ impl CopyTable {
             .enumerate()
         {
             // is_first
-            let is_first = if step_idx == 0 { F::one() } else { F::zero() };
+            let is_first = Value::known(if step_idx == 0 { F::one() } else { F::zero() });
             // is last
             let is_last = if step_idx == copy_event.bytes.len() * 2 - 1 {
-                F::one()
+                Value::known(F::one())
             } else {
-                F::zero()
+                Value::known(F::zero())
             };
 
             // id
             let id = if is_read_step {
-                number_or_hash_to_field(&copy_event.src_id, randomness)
+                number_or_hash_to_field(&copy_event.src_id, challenges.evm_word())
```

Comment:
The `id` field can contain a hash (code_hash) when the copy is from or to bytecode.  And hash is an RLC of a word.  So I think the `id` should also be `SecondPhase`